### PR TITLE
Example bootstrap sass

### PR DESCRIPTION
**Pull request by UnderTheKnife**

### CODE REVIEW

File: example_bootstrap_sass/package.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
+{
+  "name": "example_bootstrap",
+  "version": "1.0.0",
+  "description": "example bootstrap page",
+  "main": "index.html",
+  "scripts": {
+    "scss": "node-sass -o css sass/styles.scss",
+    "sass-lint": "sass-lint  sass/*.scss -v -q && echo ok || echo not ok"
+  },
+  "devDependencies": {
```
Comment (StasVysotskyi) [inline example_bootstrap_sass/package.json]: Learn more about dependencies and devDependencies in the package.json. I think bootstrap and jquery is not dev dependency in this project

File: example_bootstrap_sass/package.json [hunk 2]
```diff
@@ -0,0 +1,18 @@
+{
+  "name": "example_bootstrap",
+  "version": "1.0.0",
+  "description": "example bootstrap page",
+  "main": "index.html",
+  "scripts": {
```
Comment (StasVysotskyi) [inline example_bootstrap_sass/package.json:6]: and 'start' script for running npm server

File: example_bootstrap_sass/README.MD [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+```
+Node.js 8:
+```
+    curl -sL https://deb.nodesource.com/setup_8.x | sudo -E bash - 
+    sudo apt-get install -y nodejs
+```
+Http-server install:
+```
+     npm install http-server -g
```
Comment (StasVysotskyi) [inline example_bootstrap_sass/README.MD]: you also can save this dependency line dev dependency and run your server as ./node_modules/.bin/http-server -a localhost -p 8000. Also you can set this command in the npm start script.

File: example_bootstrap_sass/index.html [hunk 4]
```diff
@@ -0,0 +1,173 @@
+<!DOCTYPE html>
+<html lang="en">
+    <head>
+        <meta charset="UTF-8">
+        <meta name="viewport" content="wclassth=device-wclassth, initial-scale=1">
+        <title>blog</title>
+        <link rel="stylesheet" href="node_modules/bootstrap/dist/css/bootstrap.min.css">
```
Comment (StasVysotskyi) [inline example_bootstrap_sass/index.html]: please, set all links to the resource with "/" on the start of path. It's very importent. If you woun't set this char, you will have problem on the pages, where url after domain name have some string, like: http://domain.name/test/. In this case your resource will loaded from "http://domain.name/test/node_modules/..." but should be from "http://domain.name/node_modules/..."

File: example_bootstrap_sass/README.MD [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+```
+### Launching
+You must install packages from `package.json`. \
+Run on project folder:
+```
+    npm install
+```
+For generate `styles.css` file run in project folder:
+```
+    npm run scss
```
Comment (StasVysotskyi) [inline example_bootstrap_sass/README.MD:24]: Please, add you css folder (after generation) to gitignore file.

File: example_bootstrap_sass/index.html [hunk 6]
```diff
@@ -0,0 +1,173 @@
...
+        <link rel="stylesheet" href="css/styles.css">
+    </head>
+    <body>
+        <div class="container">
```
Comment (StasVysotskyi) [inline example_bootstrap_sass/index.html]: lets try to change your div's with SEO friendly HTML5 tags, like header, main, section. http://hop-online.com/blog/html5-seo-simple-coding-for-search-engine-friendliness/

File: example_bootstrap_sass/index.html [hunk 7]
```diff
@@ -0,0 +1,173 @@
...
+                    <div class="clearfix visible-xs"></div>
+                    <div class="col-md-2 col-sm-2 col-xs-10 col-md-push-4 col-sm-push-4 col-xs-push-1">
+                        <div class="form-div">
+                            <h3> NEWSLETTER</h3>
+                            <p>
+                                Lorem ipsum dolor sit amet, consectetur adipiscing elit.
+                            </p>
+                            <div class="form">
+                                <input class="email_field" type="email" placeholder="Email">
+                                <input type="submit" class="ok" value="OK">
```
Comment (StasVysotskyi) [inline example_bootstrap_sass/index.html]: If this is form, you should add <form> container to the form with action and form type. Also add required field to the email input.

File: example_bootstrap_sass/sass/styles.scss [hunk 8]
```diff
@@ -0,0 +1,13 @@
+@import 'content';
```
Comment (StasVysotskyi) [inline example_bootstrap_sass/sass/styles.scss]: In the your directory you have to use specific file name, like:
_content.scss
_header.scss
_footer.scss
It means that this files is part of your styles.

File: example_bootstrap_sass/sass/styles.scss [hunk 9]
```diff
@@ -0,0 +1,13 @@
...
+@import 'header';
+@import 'footer';
+$zero-margin: 0;
+
```
Comment (StasVysotskyi) [inline example_bootstrap_sass/sass/styles.scss]: I recommend you keep following folders in you sass directory:
./elements
./common
in the elements folder you can store all elements files of you website, like header, footer, section and etc. and in the common you can store file for your colors, fonts and other general files

File: example_bootstrap_sass/sass/header.scss [hunk 10]
```diff
@@ -0,0 +1,12 @@
+$text-color: #FEFEFE;
```
Comment (StasVysotskyi) [inline example_bootstrap_sass/sass/header.scss]: keep colors of your website in separate file

File: example_bootstrap_sass/sass/footer.scss [hunk 11]
```diff
@@ -0,0 +1,51 @@
+$footer-background: "../img/footer.png";
```
Comment (StasVysotskyi) [inline example_bootstrap_sass/sass/footer.scss]: keep colors of your website in separate file

File: example_bootstrap_sass/sass/content.scss [hunk 12]
```diff
@@ -0,0 +1,60 @@
+$line-background: "../img/line.png";
```
Comment (StasVysotskyi) [inline example_bootstrap_sass/sass/content.scss]: keep colors of your website in separate file

File: example_bootstrap_sass/sass/content.scss [hunk 13]
```diff
@@ -0,0 +1,60 @@
...
+  border-radius: 0;
+  width: 170px;
+}
+.line {
+  margin-top: 23px;
+  height: 1px;
+  background: url($line-background);
+  background-repeat: repeat-x;
+}
+.content a:link, .content a:visited {
```
Comment (StasVysotskyi) [inline example_bootstrap_sass/sass/content.scss]: you have to use tree view of your styles. Replace all of your styles  following structure

> .content {
>   a {
>     &:link {
>       ...
>     }
>   }
> }

File: example_bootstrap_sass/index.html [hunk 14]
```diff
@@ -0,0 +1,173 @@
+<!DOCTYPE html>
+<html lang="en">
+    <head>
+        <meta charset="UTF-8">
+        <meta name="viewport" content="wclassth=device-wclassth, initial-scale=1">
+        <title>blog</title>
```
Comment (StasVysotskyi) [inline example_bootstrap_sass/index.html]: add description and keywords meta tags

File: example_bootstrap_sass/README.MD [hunk 15]
```diff
@@ -0,0 +1,40 @@
...
+Then you must start the local server:
+```
+    http-server
+```
+Now you can visit http://localhost:8080 to view your server
+
+### Syntax validate
+For check syntax with `sass-lint` run in project folder:
+```
+    npm run sass-lint
```
Comment (StasVysotskyi) [inline example_bootstrap_sass/README.MD:34]: check you sass linter. I found a lot of warnings

File: example_bootstrap_sass/package.json (hunk 2 above)
Comment (UnderTheKnife) [reply example_bootstrap_sass/package.json:6]: done

File: example_bootstrap_sass/package.json (hunk 1 above)
Comment (UnderTheKnife) [reply example_bootstrap_sass/package.json]: done

File: example_bootstrap_sass/README.MD (hunk 3 above)
Comment (UnderTheKnife) [reply example_bootstrap_sass/README.MD]: done

File: example_bootstrap_sass/README.MD (hunk 5 above)
Comment (UnderTheKnife) [reply example_bootstrap_sass/README.MD:24]: done

File: example_bootstrap_sass/index.html (hunk 14 above)
Comment (UnderTheKnife) [reply example_bootstrap_sass/index.html]: done

File: example_bootstrap_sass/index.html (hunk 4 above)
Comment (UnderTheKnife) [reply example_bootstrap_sass/index.html]: done

File: example_bootstrap_sass/sass/styles.scss (hunk 8 above)
Comment (UnderTheKnife) [reply example_bootstrap_sass/sass/styles.scss]: done

File: example_bootstrap_sass/sass/header.scss (hunk 10 above)
Comment (UnderTheKnife) [reply example_bootstrap_sass/sass/header.scss]: done

File: example_bootstrap_sass/sass/footer.scss (hunk 11 above)
Comment (UnderTheKnife) [reply example_bootstrap_sass/sass/footer.scss]: done

File: example_bootstrap_sass/sass/content.scss (hunk 12 above)
Comment (UnderTheKnife) [reply example_bootstrap_sass/sass/content.scss]: done

File: example_bootstrap_sass/sass/styles.scss (hunk 9 above)
Comment (UnderTheKnife) [reply example_bootstrap_sass/sass/styles.scss]: done

File: example_bootstrap_sass/index.html (hunk 6 above)
Comment (UnderTheKnife) [reply example_bootstrap_sass/index.html]: done

File: example_bootstrap_sass/sass/content.scss (hunk 13 above)
Comment (UnderTheKnife) [reply example_bootstrap_sass/sass/content.scss]: done

File: example_bootstrap_sass/index.html (hunk 7 above)
Comment (UnderTheKnife) [reply example_bootstrap_sass/index.html]: done

File: example_bootstrap_sass/README.MD (hunk 15 above)
Comment (UnderTheKnife) [reply example_bootstrap_sass/README.MD:34]: done

**Pull request merged**